### DISCUSSION TITLE
How to combine multiple plots in one plot?

### DISCUSSION THREAD

**Discussion by FLX90** [colour-science/colour#1253] [Q&A] [answered]
Hi,

how can I plot multiple figures in one plot?
Basically I want those two figures in one plot:

colour.plotting.plot_planckian_locus_in_chromaticity_diagram_CIE1931(["A", "B", "C"],show=False);
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(["ITU-R BT.709", "ITU-R BT.2020"],show=True)

But they're appearing one after another ...
I also tried the show-parameter (formerly known as standalone and discussed here: https://github.com/colour-science/colour/issues/552). But it won't work.

Thanks for your help.

**Comment by thomasmansencal** [comment]
Hello @FLX90,

We have an example here: https://colab.research.google.com/drive/1NRcdXSCshivkwoU2nieCvC3y14fx1X4X#scrollTo=IffyjISw8Unz, the key being to pass the `axes` to our plotting definitions.

Cheers,

Thomas

**Reply by FLX90** [reply]
Hey @KelSolaar ,

Thank you for your answer.

I got both figures now on top of the other.

```
import colour
import matplotlib.pyplot as plt

figure = plt.figure()

axes = figure.add_subplot(211)
colour.plotting.plot_planckian_locus_in_chromaticity_diagram_CIE1931(["A", "B", "C"], axes=axes, show=False)
axes = figure.add_subplot(212)
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(["ITU-R BT.709", "ITU-R BT.2020"], axes=axes, show=False)

plt.show()
```

But I need them stacked together. So that I've gamut and locus togehter.

![Figure_1](https://github.com/colour-science/colour/assets/45593561/783b5402-9755-4598-9a43-0084714391af)
![Figure_2](https://github.com/colour-science/colour/assets/45593561/6df403ca-b2b8-4a30-b4ee-5e5f18ce4b98)

**Reply by thomasmansencal** [reply]
Simple as doing that:

```
import colour
import matplotlib.pyplot as plt

figure = plt.figure()

axes = figure.add_subplot(121)
colour.plotting.plot_planckian_locus_in_chromaticity_diagram_CIE1931(["A", "B", "C"], axes=axes, show=False)
axes = figure.add_subplot(122)
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(["ITU-R BT.709", "ITU-R BT.2020"], axes=axes, show=False)

plt.show()
```

See https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplot.html documentation:

```
add_subplot(nrows, ncols, index, **kwargs)
```

**Reply by FLX90** [reply]
I get them stacked together:

```
import colour
import matplotlib.pyplot as plt

figure = plt.figure()

axes = figure.add_subplot(111)
colour.plotting.plot_planckian_locus_in_chromaticity_diagram_CIE1931(["A", "B", "C"], axes=axes, show=False)
axes = figure.add_subplot(111)
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(["ITU-R BT.709", "ITU-R BT.2020"], axes=axes, show=False)

plt.show()
```

But now they overlay each other so that I can only see one figure.
I need gamut and locus in one figure.

Like here: https://github.com/colour-science/colour/issues/552

**Reply by thomasmansencal** [reply]
Hello,

You did not use the proper indexing, see my code above, it should be `121` and `122`, i.e., 1 row, 2 columns, index 1 and 2.

Cheers,

Thomas

**Reply by FLX90** [reply]
Hey,

This'll show them side by side:

![Figure_12](https://github.com/colour-science/colour/assets/45593561/68fee10c-f6b7-4038-9c6c-edb74181dadf)

**Reply by thomasmansencal** [reply, marked as answer]
Oh I understand what you want now, sorry! So you want 2 plots (not figures) in one plot, try that:

```python
import colour
import matplotlib.pyplot as plt

figure, axes = colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(["ITU-R BT.709", "ITU-R BT.2020"], show=False)
colour.plotting.temperature.plot_planckian_locus(axes=axes, show=True)

plt.show()
```

![image](https://github.com/colour-science/colour/assets/99779/2eefa566-d407-49d7-b418-3b6ed9276886)

**Reply by FLX90** [reply]
Yes, that's exactly what I was looking for.
Thank you!

**Reply by thomasmansencal** [reply]
Awesome and glad to hear!
